src/explorer/assets/refresh.js

```diff
@@ -0,0 +1,25 @@
+// Mimi UI
+
+function refresh(speclist) {
+    
+    // Loop over the things that we want to show in the list, add
+    // one button for each element as a child to the variablelist div
+
+    for (var i in speclist) {
+
+        var newButton = document.createElement("button");
+        newButton.setAttribute("class", "tab");
+        
+        // Return a closure with a copy of the spec that's private to the function
+        newButton.onclick = (function() {
+            var spec = speclist[i]["VLspec"];
+
+            return function() {
+                vegaEmbed("#vis", spec, {actions: false});
+            }
+        }())
```

Comment:
This seems unnecessary complicated. Can't this be just
````javascript
var spec = speclist[i]["VLspec"];
newButton.onclick = function() {
    vegaEmbed("#vis", spec, {actions: false});
}

Reply:
@rjplevin this was the result of the debugging work we did to make sure each graph rendered on its own and did not write over the previous ones ... would this simplification work?

Reply:
We tried the approach you suggested, but the value of spec changes between calls, so all the functions end up referencing the same (final) spec. Using a closure solved this problem.

Reply:
Thanks, that's what I thought, so we'll leave it for now.

Reply:
Hm, really strange, but yes, lets just leave as is if it works :)